### DISCUSSION TITLE
Stream randomly aborted due to Runtime Error

### DISCUSSION THREAD

**Discussion by juliacollm** [bonsai-rx/bonsai#1473] [Q&A] [answered]
Hello :)

I have a Bonsai workflow that controls two photometry systems (FP3002) and receives TTL input from two breakout boxes. The system has been running quite smoothly for months but lately my recordings are being interrupted due to a Runtime Error. This happens randomly sometime after having started the workflow (sometimes is only 5 minutes after, others is after 1 hour, and others somewhere in between). 
This is the message I get when the recording is interrupted 'Spinnaker: Stream has been aborted.[-1012 ]' (see picture1)

When this error occurs I have to shut down Bonsai and the two photometry systems completely, otherwise when restarting the workflow I get different Runtime errors, but always related to the Spinnaker (SpinView 1.29.0.5) (see pictures 2 and 3).

In the several months I've been using this system I've encountered this issue only a couple of times, but for the last two weeks is happening multiple times a day. Nothing has really changed besides that the system was unused for 2 weeks. Has anyone encountered this problem or has suggestions to how to solve it?

Many thanks! 

![picture3](https://github.com/bonsai-rx/bonsai/assets/139750762/59232843-b68d-4c27-bf88-dd8417d6cf92)
![picture1](https://github.com/bonsai-rx/bonsai/assets/139750762/6eb160a4-7cab-4331-a3f4-6d5166a54ce5)
![picture2](https://github.com/bonsai-rx/bonsai/assets/139750762/147f9eec-5206-40de-9cf7-c97b2897ea50)

**Comment by bruno-f-cruz** [comment]
The second error seems to be related to attempting to write a specific combination of settings that the camera does not support. What node are you trying to use to interface with the camera? If it is specific to this package, it might be better to ask the [package repository directly.](https://github.com/neurophotometrics/neurophotometrics/tree/master/Neurophotometrics).

It might also be helpful to include the version of the spinnaker package you are using and the SDK version you have installed.

**Reply by juliacollm** [reply]
SpinView 1.29.0.5 is the one I'm using. 

And here I attach a picture of how the node tree looks. 

Thanks for the fast reply!

![photo1689608447](https://github.com/bonsai-rx/bonsai/assets/139750762/058585b3-5e18-4464-b8ef-aabd09b3164b)

**Reply by bruno-f-cruz** [reply]
Do these errors occur if you use the latest available bonsai.spinnaker package? Using SpinnakerCapture source node.

**Reply by juliacollm** [reply]
This is the version of the package. I don't know if it's the latest one. 
![msg420332135-130683](https://github.com/bonsai-rx/bonsai/assets/139750762/d14f6f99-64e9-471c-b613-4823b06ebcc7)

**Reply by bruno-f-cruz** [reply]
And the camera crashes using the spinnacapture source node?

**Reply by juliacollm** [reply]
I've never used this node, I don't even now how to set it up. 

At the moment I'm totally unable to start any Bonsai workflow with any on my 3 FP3002 devices (in different setups). SpinView is also giving warning messages and when I try to start the camera I get this (picture)

![msg420332135-130689](https://github.com/bonsai-rx/bonsai/assets/139750762/97e245c1-4645-46c7-b5e8-02b63c880104)


I've tried to update the device firmware but just made it worse, cause now it says the device is not compatible. I'm totally clueless about what's going on.

**Reply by glopesdev** [reply]
@juliacollm Have you tried reaching out to Neurophotometrics support team directly? They should have more experience on hardware firmware version compatibility and also specifically for troubleshooting FP3002 systems.

**Reply by juliacollm** [reply, marked as answer]
Thanks for all the replies! Indeed Neurophotometrics had the answer, it was a **hardware** issue. 

I'll leave a brief explanation in case is useful for someone facing the same errors: Some bent pins in the USB 3 to USB cable that connects my photometry device with the computer were causing an interruption between the communication of the two. If the communication could not be established from the beginning I was getting errors when calibrating or when starting the workflow. If the communication was lost after I started the workflow I was getting the 'stream has been aborted' error. 

So if you are getting similar errors as shown in the pictures of the first post in this discussion, try checking if there are bent pins in your device to PC connectors and ports. :)
